netlink-packet-wireguard/src/nlas/mod.rs

```diff
@@ -5,3 +5,25 @@ mod peer;
 pub use allowedip::WgAllowedIpAttrs;
 pub use device::WgDeviceAttrs;
 pub use peer::WgPeerAttrs;
+
+use netlink_packet_utils::{Emitable, nla::Nla};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) struct NestedSlice<'a, T: Nla>(&'a [T]);
+impl<'a, T: Nla> Nla for NestedSlice<'a, T> {
+    fn value_len(&self) -> usize {
+        self.0.buffer_len()
+    }
+
+    fn kind(&self) -> u16 {
+        0
```

Comment:
If I understand correctly, `NestedSlice` is intended to be used in various places. We can't know for sure that the NLA type will be 0 everytime, can we?